### DISCUSSION TITLE
Template in ModalContainer

### DISCUSSION THREAD

**Discussion by cooligus** [FrameMuse/react-modal-global#80] [Q&A]
**Describe the bug**
When using template prop it's in ModalContainer modal is always rendered.

I have a layout:
```tsx
const ModalLayout = (props: any) => {
  return (
    <>
        <div className={classes.background} onClick={() => Modal.closeAll()}></div>,
        <div className={classes.modal}>
          <div>
            <p onClick={() => {Modal.closeAll()}} className={classes.closeModal}>
              <X />
            </p>
          </div>
          <div>
            {props.children}
          </div>
        </div>
    </>
  );
};
```

And I have a ModalContainer:
```tsx
<ModalContainer template={ModalLayout}/>
```

It always renders without triggering any opens.

**Comment by FrameMuse** [comment]
This is by design. This is for your probable animations and transitions. You need to set `display: none` when popup is not active.
Take a look at the example
https://github.com/FrameMuse/react-modal-global/blob/3d1a689dc6d3227cc26c056d29c0add981354e1b/styles/modal.scss#L25

You can find more examples here
https://github.com/FrameMuse/react-modal-global#playgrounds

If I didn't quite understand you, tell me more about your case.

**Comment by FrameMuse** [comment]
@cooligus Could you give more information on it?

**Comment by cooligus** [comment]
I have realised that CSSModules can have problem handling custom classes.

https://github.com/FrameMuse/react-modal-global/blob/3d1a689dc6d3227cc26c056d29c0add981354e1b/styles/modal.scss#L25
I think it would be very powerful if we could write something like:
```tsx 
<ModalContainer template={ModalLayout} activeClass="activeClass"}>
```
Nice to have.

**Comment by FrameMuse** [comment]
This code is in SCSS
[react-modal-global/styles/modal.scss](https://github.com/FrameMuse/react-modal-global/blob/3d1a689dc6d3227cc26c056d29c0add981354e1b/styles/modal.scss#L25)

You can do the same by typing `.modal--active { ... }` in plain CSS.

**Comment by cooligus** [comment]
Problem of this issue is: class `.modal--active` is in component with template all the time.

**Reply by FrameMuse** [reply]
So you have `.modal--active` all the time, no matter what, right?

**Reply by FrameMuse** [reply]
Could you provide a code example of your implementation on sandbox?

**Comment by cooligus** [comment]
Sorry you had to wait so long.

Let's say it's our App.tsx:
```tsx
import { Modal, ModalContainer } from "react-modal-global";
import { ReactNode } from "react";
import { useModalContext } from "react-modal-global";

const Example = () => {
  return <>HELLLO</>;
};

const PopupLayout = (props: { children: ReactNode }) => {
  return (
    <div>
      <div>Example management</div>
      <div>{props.children}</div>
    </div>
  );
};

function App() {
  //Modal.open(Example);
  return (
    <div className="App">
      <ModalContainer template={PopupLayout} />
    </div>
  );
}

export default App;
```
And this is the result:
![image](https://user-images.githubusercontent.com/67282004/229595391-9e8dd086-17ec-4fa0-93d9-979bb02b578e.png)

Anyway I use 1.2.1 version of the library and react still screams that useModalContext must be used within its context.

**Reply by FrameMuse** [reply]
In this example, `useModalContext` is not used anywhere.
To make a modal disappear by default, you should add styles, like this:
```
import "react-modal-global/styles/modal.scss";
```

Styles are provided by the library, but you can write your own by the example here:
https://github.com/FrameMuse/react-modal-global/blob/3d1a689dc6d3227cc26c056d29c0add981354e1b/styles/modal.scss#L25

**Reply by FrameMuse** [reply]
I created an example using your code
https://codesandbox.io/s/react-modal-global-examples-forked-xsj2kv

**Comment by cooligus** [comment]
Thanks bro for example!

(Any further reader reading it (add to readme info that dev need to add their own classes))
(useModalExample had nothing to do with this code)
